Tests/DatadogTests/Datadog/Mocks/SystemFrameworks/FoundationMocks.swift

```diff
@@ -412,6 +418,13 @@ extension URLRequest: AnyMockable {
         request.httpMethod = httpMethod
         return request
     }
+
+    static func mockWith(url: String, queryParams: [URLQueryItem], httpMethod: String) -> URLRequest {
+        let url: URL = .mockWith(url: url, queryParams: queryParams)
+        var request = URLRequest(url: url)
+        request.httpMethod = httpMethod
+        return request
+    }
```

Comment:
We could simplify mocking by getting rid of `func mockWith(url: String, queryParams: [URLQueryItem]) -> URL`

and do
```swift
static func mockWith(url string: String, httpMethod: String = "GET") -> URLRequest {
    let url: URL = URL(string: string)!
    var request = URLRequest(url: url)
    request.httpMethod = httpMethod
    return request
}

let request: URLRequest = .mockWith(url: "https://www.example.com?foo=42&lang=en")
```
WDYT?

Reply:
I'm always more inclined to have more control. Also, assuming we could use some fuzzy testing in iOS, it'd be easier to generate a random base url and a random list of query param to build the full url than writing all like you propose